### ISSUE TITLE
Generic.Functions.FunctionCallArgumentSpacing fails to fix `foo('bar',);` when PSR2.Methods.FunctionCallSignature is used

### ISSUE THREAD

**Issue by mabar** [squizlabs/PHP_CodeSniffer#3707]
**Describe the bug**

`Generic.Functions.FunctionCallArgumentSpacing.NoSpaceAfterComma` should not report in case comma is followed by right parenthesis because it collides with `PSR2.Methods.FunctionCallSignature.SpaceBeforeCloseBracket` and fails to fix the file

**Code sample**

```php
foo('bar',);
```

**Custom ruleset**
```xml
<?xml version="1.0"?>
<ruleset>
  <rule ref="Generic.Functions.FunctionCallArgumentSpacing"/>
  <rule ref="PSR2.Methods.FunctionCallSignature"/>
</ruleset>
```

**To reproduce**
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs test.php ...`
3. See error message displayed

```
No space found after comma in argument list (Generic.Functions.FunctionCallArgumentSpacing.NoSpaceAfterComma)
```

**Expected behavior**
No error.

**Versions:**
 - OS: Windows 10
 - PHP: 8.1.9
 - PHPCS: 3.7.1

**Comment by jrfnl** [comment]
This issue has been fixed via PR https://github.com/squizlabs/PHP_CodeSniffer/pull/3805. The fix will be included in PHPCS 3.8.0.

**Comment by jrfnl** [comment]
FYI: the fix for this issue is included in today's [PHP_CodeSniffer 3.8.0 release](https://github.com/PHPCSStandards/PHP_CodeSniffer/releases/tag/3.8.0).

As per #3932, development on PHP_CodeSniffer will continue in the [PHPCSStandards/PHP_CodeSniffer](https://github.com/PHPCSStandards/PHP_CodeSniffer) repository. If you want to stay informed, you may want to start "watching" that repo (or watching releases from that repo).
